tools: improve label preview

MylesBorins opened the pull request:
* change label to `create-preview`
* update labels throughout progress to give status
* add label if job fails
* fix error with remove label

MylesBorins commented:
I think we need to land this before we can test it

MylesBorins commented:
one thing I'm noticing with the labelling action is that we are now ending up with a "label to staging" skipped action after any labels are added to the repo, which is a bit less than ideal.

/cc @mmarchini were you having this issue on nodejs/node?

nschonni commented on .github/workflows/generate-preview.yml:
```diff
@@ -29,6 +29,30 @@ jobs:
+              name: ['create-preview']
+            })
+
+      - name: Add generating-preview label
```
I'd probably skip this part, since even though it will no-op, it will show up in the build results like on this preview as an extra "cancelled" action. Not sure if that result might also override the result of this successful run

nschonni commented on .github/workflows/generate-preview.yml:
```diff
@@ -57,14 +81,27 @@ jobs:
               owner: context.repo.owner,
               repo: context.repo.repo,
-              name: ['start-ci']
+              name: ['preview-failed']
```
I think the signature is a little different from the remove
```suggestion
              labels: ['preview-failed']
```

nschonni commented on .github/workflows/generate-preview.yml:
```diff
@@ -29,6 +29,30 @@ jobs:
+              issue_number: github.event.pull_request.number,
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              name: ['create-preview']
```
Might have messed this up in the original version, I'm not sure this is an array or just a single string

nschonni commented on .github/workflows/generate-preview.yml:
```diff
@@ -29,6 +29,30 @@ jobs:
+              issue_number: github.event.pull_request.number,
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              name: ['create-preview']
```
I think the signature is a little different from the remove. If you decide to keep this step
```suggestion
              labels: ['create-preview']
```

nschonni commented:
Sorry, it looks like removing the action/cache created a conflict

MylesBorins commented:
@nschonni already fixed it locally :P

MylesBorins commented:
@nschonni I've updated the label commands based on the docs. Removed the building and failure labels since there will we definitely get feedback in the UI for this.

nschonni commented:
I think the expression for the createComment might also need to be fixed like you did for the label

mmarchini commented:
> one thing I'm noticing with the labelling action is that we are now ending up with a "label to staging" skipped action after any labels are added to the repo, which is a bit less than ideal.

Yes, but if the label differs from `create-preview` it should skip

> /cc @mmarchini were you having this issue on nodejs/node?

No because we use a different event (schedule)

nschonni commented:
Looks like it doesn't like the `number` part still https://github.com/nodejs/nodejs.dev/pull/869/checks?check_run_id=986711290#step:4:25

mmarchini commented:
Try `${{ github.event.number }}` (either directly on the script or as an environment variable which can be used in the script with `process.env`).
